dm/syncer/checkpoint.go

```diff
@@ -561,12 +685,14 @@ func (cp *RemoteCheckPoint) FlushPointsExcept(tctx *tcontext.Context, exceptTabl
 		return err
 	}
 
-	cp.globalPoint.flush()
-	for _, point := range points {
-		point.flush()
+	if snapshotCp.globalPoint != nil {
+		cp.globalPoint.flushBy(*snapshotCp.globalPoint)
+		cp.globalPointSaveTime = snapshotCp.globalPointSaveTime
 	}
 
-	cp.globalPointSaveTime = time.Now()
+	for _, point := range points {
+		point.tableCp.flushBy(point.snapshotTableCP)
+	}
 	cp.needFlushSafeModeExitPoint = false
```

Comment:
golang's race detector will report `needFlushSafeModeExitPoint` is written by event loop of syncer 
https://github.com/pingcap/ticdc/blob/6760414c14168950fda9525e1f299adb3699a235/dm/syncer/syncer.go#L1806
and checkpointFlushWorker's goroutine (though a bool should be ok to race). We might use Atomic.Bool